src/main/java/net/imagej/ops/image/distancetransform/DistanceTransform3D.java

```diff
@@ -0,0 +1,305 @@
+package net.imagej.ops.image.distancetransform;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+
+import net.imagej.ops.Contingent;
+import net.imagej.ops.Ops;
+import net.imagej.ops.create.img.CreateImgFromDimsAndType;
+import net.imagej.ops.special.function.Functions;
+import net.imagej.ops.special.function.UnaryFunctionOp;
+import net.imagej.ops.special.hybrid.AbstractUnaryHybridCF;
+import net.imglib2.FinalInterval;
+import net.imglib2.RandomAccess;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.type.BooleanType;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.type.numeric.real.FloatType;
+
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.thread.ThreadService;
+
+/**
+ * @author Simon Schmid (University of Konstanz)
+ */
+@Plugin(type = Ops.Image.DistanceTransform.class)
+public class DistanceTransform3D<B extends BooleanType<B>, T extends RealType<T>>
+		extends AbstractUnaryHybridCF<RandomAccessibleInterval<B>, RandomAccessibleInterval<T>>
+		implements Ops.Image.DistanceTransform, Contingent {
+
+	@Parameter
+	private ThreadService ts;
+
+	@SuppressWarnings("rawtypes")
+	private UnaryFunctionOp<FinalInterval, RandomAccessibleInterval> createOp;
+
+	private ExecutorService es;
+
+	@Override
+	public boolean conforms() {
+		long max_dist = in().dimension(0) * in().dimension(0) + in().dimension(1) * in().dimension(1)
```

Comment:
same comments as in `2D` case. Here you should first check if `in()` is actually 3-dimensional.